File: 2075-decode-the-slanted-ciphertext/2075-decode-the-slanted-ciphertext.cpp
class Solution {
public:
    string decodeCiphertext(string s, int rows) {
        int n = s.size(), cols = n / rows;
        string res;
        for (int i = 0; i < cols; ++i)
            for (int j = i; j < n; j += cols + 1)
                res += s[j];
        while (res.back() == ' ')
            res.pop_back();
        return res;
    }
};

File: 2075-decode-the-slanted-ciphertext/2075-decode-the-slanted-ciphertext.py
class Solution(object):
    def decodeCiphertext(self, encodedText, rows):
        if rows == 1:
            return encodedText  
        n = len(encodedText)
        cols = n // rows 
        res = []
        for start_col in range(cols):
            r, c = 0, start_col
            while r < rows and c < cols:
                res.append(encodedText[r * cols + c])
                r += 1
                c += 1
        return "".join(res).rstrip()